src/constants.ts

```diff
@@ -24,9 +33,68 @@ export type BaseSystemFields<T = never> = {
 \texpand?: T
 }`
 
+export const BASE_SYSTEM_CREATE_FIELDS_DEFINITION = `export type BaseSystemCreateFields = {
+\tid?: ${RECORD_ID_STRING_NAME}
+}`
+
+export const BASE_SYSTEM_UPDATE_FIELDS_DEFINITION = `export type BaseSystemUpdateFields = unknown`
+
 export const AUTH_SYSTEM_FIELDS_DEFINITION = `export type AuthSystemFields<T = never> = {
 \temail: string
 \temailVisibility: boolean
 \tusername: string
 \tverified: boolean
 } & BaseSystemFields<T>`
+
+export const AUTH_SYSTEM_CREATE_FIELDS_DEFINITION = `export type AuthSystemCreateFields = {
+\tid?: ${RECORD_ID_STRING_NAME}
+\temail: string
+\temailVisibility?: boolean
+\tpassword: string
+\tpasswordConfirm: string
+\tverified?: boolean
+}`
+
+export const AUTH_SYSTEM_UPDATE_FIELDS_DEFINITION = `export type AuthSystemUpdateFields = {
+\temail?: string
+\temailVisibility?: boolean
+\toldPassword?: string
+\tpassword?: string
+\tpasswordConfirm?: string
+\tverified?: boolean
+}`
```

Comment:
Remove the 4 constants above (BASE_SYSTEM_CREATE_FIELDS_DEFINITION, BASE_SYSTEM_UPDATE_FIELDS_DEFINITION, AUTH_SYSTEM_CREATE_FIELDS_DEFINITION, AUTH_SYSTEM_UPDATE_FIELDS_DEFINITION)

and inline the definitions in `CreateAuth` etc below.